core/src/G3Reader.cxx

```diff
@@ -68,12 +70,18 @@ void G3Reader::Process(G3FramePtr frame, std::deque<G3FramePtr> &out)
 		out.push_back(frame);
 	}
 
+	// For python threaded contexts, release the GIL while
+	// blocking for data (which can occur on stream.peek as well
+	// as load(stream_)).  All paths out of this function must
+	// reacquire the lock!
+	PyThreadState *_save = PyEval_SaveThread();
```

Comment:
Is there a way here to tell if we are running with a Python interpreter?

Reply:
Yes, there sure is... fixed in https://github.com/CMB-S4/spt3g_software/pull/14/commits/23412a77a26b3ff80708447a8fc209b72057dc33.